ata_pipeline0/helpers/preprocessors.py

```diff
@@ -163,3 +179,22 @@ def transform(self, df: pd.DataFrame) -> pd.DataFrame:
 
     def log_result(self, df_in=None, df_out=None) -> None:
         logger.info(f"Added site name {self.site_name} as a new field")
+
+
+@dataclass
+class ReplaceNaNs(Preprocessor):
+    """
+    Replaces all `np.nan` instances in the DataFrame with a specified alternative.
+    """
+
+    replace_with: Any
+
+    def transform(self, df: pd.DataFrame) -> pd.DataFrame:
+        df = df.copy()
```

Comment:
🔧 Don't think we need this line since `df.replace` doesn't modify the DataFrame in-place but returns a new DataFrame by default. I only do `df.copy` if I need perform some in-place transformation, e.g., changing the data type of individual columns.